ssl/ssl_key_share.cc

```diff
@@ -294,6 +295,233 @@ class CECPQ2KeyShare : public SSLKeyShare {
   HRSS_private_key hrss_private_key_;
 };
 
+class KEMKeyShare : public SSLKeyShare {
+ public:
+  KEMKeyShare(int nid, uint16_t group_id) : nid_(nid), group_id_(group_id) {}
+
+  uint16_t GroupID() const override { return group_id_; }
+
+  bool Offer(CBB *out) override {
+    // Ensure |out| is valid, has no children, and has been initialized.
+    // We check that |out| has no children because otherwise CBB_data()
+    // will produce a fatal error by way of assert(cbb->child == NULL).
+    if (!out || out->child || !CBB_data(out)) {
+      OPENSSL_PUT_ERROR(SSL, ERR_R_PASSED_NULL_PARAMETER);
+      return false;
+    }
+
+    // ctx_ should not have been initialized (by e.g. a previous call
```

Comment:
why not? The EVP API is designed such that a single `EVP_PKEY_CTX` can be used for multiple key generations

Reply:
See above